bigquery/google/cloud/bigquery/client.py

```diff
@@ -1531,6 +1531,29 @@ def load_table_from_dataframe(
         if location is None:
             location = self.location
 
+        # If table schema is not provided, we try to fetch the existing table
+        # schema, and check if dataframe schema is compatible with it.
+        if not job_config.schema:
+            try:
+                table = self.get_table(destination)
+            except google.api_core.exceptions.NotFound:
+                table = None
+            else:
+                table_col_names = {field.name for field in table.schema}
+                dframe_col_names = set(dataframe.columns)
+
+                in_dframe_only = dframe_col_names - table_col_names
```

Comment:
And I was actually trying to avoid an unnecessary roundtrip to the server. :)

On a second thought, preferring server errors reduces the risk of the client getting out of sync with it in egde cases, and avoids logic duplication.

Reply:
> Should we set ALLOW_FIELD_ADDITION if no schema_update_options has been supplied?

What is the current expected behavior in a "typical" use case? Do users expect (and want) implicit schema updates?

I suppose a typo in dataframe column name should _not_ result in a new column in the table, unless ALLOW_FIELD_ADDITION is explicitly set by the user.